mir_planning/mir_actions/mir_manipulate_drawer/ros/scripts/open_drawer_server

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import tf
+import copy
+import rospy
+import smach
+
+# import of generic states
+import mir_states.common.manipulation_states as gms
+import mcr_states.common.basic_states as gbs
+import mir_states.common.action_states as gas
+
+# action lib
+from smach_ros import ActionServerWrapper, IntrospectionServer
+
+from std_msgs.msg import Float32MultiArray
+from geometry_msgs.msg import PoseStamped, Quaternion
+from mir_planning_msgs.msg import GenericExecuteAction, GenericExecuteResult, GenericExecuteFeedback
+from mir_actions.utils import Utils
+from mir_manipulate_drawer.manipulate_drawer_utils import ManipulateDrawerUtils
+from mir_manipulate_drawer.manipulate_drawer_utils import Setup, SendSafePoseToDBC, SendPoseToDBC
+        
+#===============================================================================
+
+class AddPoseToUserdata(smach.State):
+    def __init__(self, topic_name):
+        smach.State.__init__(self, outcomes=['succeeded', 'timeout'],
+                                   input_keys=['handle_pose'],
+                                   output_keys=['handle_pose'])
+        _pose_sub = rospy.Subscriber(topic_name, PoseStamped, self._pose_cb)
+        self._pose = None
+        rospy.sleep(0.1) # time for the publisher to register in ros network
+
+    def _pose_cb(self, msg):
+        self._pose = msg
+
+    def execute(self, userdata):
+        if self._pose is not None:
+            userdata.handle_pose = self._pose
+            print("\nadding pose to userdata\n")
+            rospy.loginfo(self._pose)
+            self._pose = None
+            return 'succeeded'
+        else:
+            rospy.loginfo("Timeout")
+            return 'timeout'
+
+#===============================================================================
+
+class SendPoseToMoveIt(smach.State):
+    def __init__(self):
+        smach.State.__init__(self, outcomes=['succeeded'],
+                                   input_keys=['handle_pose'])
+        self._moveit_pose_pub = rospy.Publisher('/moveit_to_pose/pose_in',
+                                                PoseStamped, queue_size=10)
+        self.utils = ManipulateDrawerUtils()
+        rospy.sleep(0.1) # time for the publisher to register in ros network
+
+    def execute(self, userdata):
+        arm_target_pose = copy.deepcopy(userdata.handle_pose)
+        arm_target_pose.header.stamp = rospy.Time.now()
+        arm_target_pose.pose.position.x += -0.005
+        arm_target_pose.pose.position.y += 0.0
+        arm_target_pose.pose.position.z += 0.015
```

Comment:
this needs to use pregrasp planner eventually

Reply:
https://github.com/b-it-bots/mas_industrial_robotics/issues/104